rotkehlchen/db/dbhandler.py

```diff
@@ -1180,22 +1186,35 @@ def edit_blockchain_accounts(
         - All accounts exist in the DB
         """
         # Update the blockchain account labels in the DB
-        bindings = [
-            (entry.address, entry.chain.value, entry.label)
-            for entry in account_data if entry.label is not None
-        ]
-        if len(bindings) > 0:
+        bindings_to_update, bindings_to_delete = [], []
+        for entry in account_data:
+            if entry.label not in [None, '']:
+                bindings_to_update.append((entry.address, entry.chain.value, entry.label))
+            else:
+                bindings_to_delete.append((entry.address, entry.chain.value))
+
+        modified_count = 0
```

Comment:
it might be better to use a deleted_count & modified_count to make debugging easier.